### DISCUSSION TITLE
How to turn off GitHub Actions for a repo?

### DISCUSSION THREAD

**Discussion by justinclift** [community/community#25591] [Actions]
<p>Looking through the GitHub Actions info for some of my repos, it appears to be completely irrelevant to most of them.</p>
<p>Unfortunately, there doesn’t seem to be an obvious way to turn it off (eg in Settings), and keywords like “disable” have no matches when searching the help docs.</p>
<p>How can Actions be turned off for a repo, so the tab doesn’t even appear?</p>
<p>Was expecting an option in Settings, like where the “Issues” checkbox and others are.</p>

**Comment by andymckay** [comment, marked as answer]
<p>If you go to “Settings” for the repository, there should be a entry on left hand menu for “Actions”. Click on that and select “Disable Actions for this repository”.</p>
<p>That should turn off the Actions tab for you.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/6/6e57cd1928057b86512b334a895cf94b594d2ece.png" data-download-href="/uploads/short-url/fK8DmtdBSUi1mbSl3diCdxl16OO.png?dl=1" title="Screen Shot 2019-11-13 at 11.18.51 AM.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/74699/181091931-2b1e66a7-74bb-4d51-8a42-5a3d3b406fa9.png" alt="Screen Shot 2019-11-13 at 11.18.51 AM.png" data-base62-sha1="fK8DmtdBSUi1mbSl3diCdxl16OO" width="690" height="311" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/6/6e57cd1928057b86512b334a895cf94b594d2ece_2_690x311.png, https://aws1.discourse-cdn.com/github/optimized/2X/6/6e57cd1928057b86512b334a895cf94b594d2ece_2_1035x466.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/6/6e57cd1928057b86512b334a895cf94b594d2ece.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/6/6e57cd1928057b86512b334a895cf94b594d2ece_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screen Shot 2019-11-13 at 11.18.51 AM.png</span><span class="informations">1228×554 63.1 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by justinclift** [comment]
<p>Thanks <a class="mention" href="/andymckay">@andymckay</a>, that works. :slight_smile:</p>

**Comment by fabiospampinato** [comment]
Does anyone know the API endpoint for doing this programmatically?

**Reply by woss** [reply]
Here is the REST api example 

- endpoint `https://api.github.com/repos/USERNAME/REPO/actions/permissions`
- method PUT
- payload `{"enable":false}`

**Comment by wasifali758595** [comment]
To turn off GitHub Actions for a repository, you have a few options depending on how much control you want:

1. Disable GitHub Actions entirely for the repo

Go to your repository on GitHub.

Click Settings (you need admin access).

In the left sidebar, select Actions → General.

Under Actions permissions, choose Disable Actions for this repository.

This stops all workflows from running.

2. Restrict Actions instead of fully disabling

In the same Settings → Actions → General page, you can limit which Actions are allowed (e.g., only those in your repo or only from verified creators).

3. Disable specific workflows only

Go to the repo’s Actions tab.

Select the workflow you want to disable.

Click the "..." (More options) button in the top-right.

Choose Disable workflow.

✅ If you just don’t want CI/CD to run temporarily, disabling individual workflows is usually the best choice.
✅ If you want to completely stop Actions in that repo, use the full disable option in Settings.
